packages/plugin-npm-cli/sources/commands/npm/publish.ts

```diff
@@ -102,11 +106,32 @@ export default class NpmPublishCommand extends BaseCommand {
         const buffer = await miscUtils.bufferStream(pack);
 
         const gitHead = await npmPublishUtils.getGitHead(workspace.cwd);
+
+        let provenance = false;
+        if (workspace.manifest.publishConfig && `provenance` in workspace.manifest.publishConfig) {
+          provenance = Boolean(workspace.manifest.publishConfig.provenance);
+          if (provenance) {
+            report.reportInfo(null, `Generating provenance statement because \`publishConfig.provenance\` field is set.`);
+          } else {
+            report.reportInfo(null, `Skipping provenance statement because \`publishConfig.provenance\` field is set to false.`);
+          }
+        } else if (this.provenance) {
+          provenance = true;
+          report.reportInfo(null, `Generating provenance statement because \`--provenance\` flag is set.`);
+        } else if (configuration.get(`npmPublishProvenance`)) {
+          provenance = true;
+          report.reportInfo(null, `Generating provenance statement because \`npmPublishProvenance\` setting is set.`);
+        } else if (process.env.NPM_CONFIG_PROVENANCE) {
+          provenance = true;
+          report.reportInfo(null, `Generating provenance statement because \`NPM_CONFIG_PROVENANCE\` env var is set.`);
+        }
```

Comment:
Let's remove this one - we already support `YARN_NPM_PUBLISH_PROVENANCE`, and we don't typically read any NPM-specific variable to configure Yarn.

Reply:
Oooh, `YARN_...` automatically maps to `.yarnrc.yml` settings? This is cool!

I'll remove this branch